File: topics/arrays/exercises/median.c
int median (const int data[], int sz)
{
    int start_idx, end_idx;
    
    start_idx = 0;
    end_idx = sz - 1; // index of last element
    
    while (start_idx != end_idx && start_idx < end_idx)
    {
        start_idx++;
        end_idx--;
    }
    
    if (start_idx == end_idx) 
        return data[start_idx];
    else 
        return data[start_idx] + data[end_idx] / 2;
}

File: topics/arrays/exercises/Median.pas
function Median (const data: array of Integer): Integer;
var
    startIdx, endIdx: Integer;
begin
    startIdx := Low(data);
    endIdx := High(data);
    
    while (startIdx <> endIdx) and (startIdx < endIdx) do
    begin
        startIdx := startIdx + 1;
        endIdx := endIdx - 1;
    end;
    
    if startIdx = endIdx then result := data[startIdx]
    else result := data[startIdx] + data[endIdx] / 2;
end;